### DISCUSSION TITLE
Chapter 14 Neural networks error report

### DISCUSSION THREAD

**Discussion by li0510** [OTexts/Discussion#77] [Error report]
1. In chapter 14.5, We try to minimise the difference between the ground truth and the forecast by using a "loss function", which is our optimization objective
```
for here, I suggest the phrase "loss function" might be "Gradient Descent" because that is the procedure that iteratively minimizes the loss function to learn (or estimate) the compressed representation of input data
```
2. In chapter 14.5, Probabilistic optimization objectives part, "we can also generate **_probabilisic_** forecasts"
```
for here, the word "probabilisic" might be "probabilistic"
```
3. In chapter 14.5, Probabilistic optimization objectives part, the code snippet, which is used for plotting "Figure 14.7: AirPassengers NHITS trained with Gaussian DistributionLoss", is missing; below is modified based on previous code snippets:
```
_, axes = plt.subplots(nrows=2, ncols=1)
fig = plot_series(df=Y_train_df, 
  forecasts_df=forecasts, level=[80,90], models=['NHITS'],
  ax=axes, rm_legend=False)
for ax in fig.axes:
  ax.set_xlabel("")
  ax.set_ylabel("")
  ax.tick_params(axis='both')
  for label in ax.get_xticklabels():
    label.set_rotation(0)
fig.suptitle("Number of passengers for different airlines")
fig.supylabel("Passengers")
fig.supxlabel("Month [1M]")
fig 
```
4. In chapter 14.7, Hyperparameter optimization, the first part of code snippets has several type errors: 
```
# We optimize for 10 steps   
-> here might be 12 steps

# i.e. we evaluate 10 different hyperparameter configurations.
model = AutoNHITS(h=12, num_samples=1)   
->here missing the argument: config=nhits_config, and num_samples=10
```

**Comment by robjhyndman** [comment]
Thanks

1. Wording improved
2. Fixed
3. Code now exposed
4. Fixed
